File: UIAutomation/Cucumber/src/test/java/Case/pages/HomePage.java
package Case.pages;

import Case.utils.Driver;
import Case.utils.Helper;
import Case.utils.Log4j;
import org.junit.Assert;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

import java.text.Format;
import java.util.List;
import java.util.Random;

public class HomePage extends Helper {
    private final Random ran;

    public HomePage() {
        PageFactory.initElements(Driver.get(), this);
        ran = new Random();
    }

    @FindBy(xpath = "//div[@id='cookieseal-banner']//button[text()='Tüm Çerezlere İzin Ver']")
    public WebElement tumCerezlereIzinVer;
    @FindBy(xpath = "//div[@class='mega-menu ']//ancestor::li[@class='menu-header-item  ']")
    public List<WebElement> kategoriMenuSecim;
    @FindBy(xpath = "//li[@class='menu-header-item menu-header-item--active ']//ul[@class='zones-lists-items']//a")
    public List<WebElement> kategoriSecim;
    @FindBy(css = ".main-header-logo")
    public WebElement anaSayfaLogo;
    @FindBy(css = ".full-width-slider.full-width-slider--horizontal ")
    public WebElement anaSayfaSlaytResmi;

    public void tumCerezlereIzinVer() {
        clickElement(tumCerezlereIzinVer);
        Log4j.info("Tüm çerezlere izin ver butonuna tıklandı.");
    }

    public void herhangiBirKategoriMenuSecim() {
        int random = ran.nextInt(kategoriMenuSecim.size());
        String kategoriMenu = kategoriMenuSecim.get(random).getText();

        hoverElement(kategoriMenuSecim.get(random));
        Log4j.info(String.format("Header bölümündeki menülerden '%s' fareyle üzerine gelindi.", kategoriMenu));
    }

    public void herhangiBirKategoriSecim() {
        int random = ran.nextInt(kategoriSecim.size());
        String kategori = kategoriSecim.get(random).getText();

        clickElement(kategoriSecim.get(random));
        Log4j.info(String.format("Header bölümündeki kategorilerden '%s' tıklandı.", kategori));
    }

    public void anaSayfayaGit() {
        clickElement(anaSayfaLogo);
        Log4j.info("Header bölümündeki logoya tıklanarak ana sayfaya gidildi.");
    }

    public void anaSayfaAcildiMi() {
        if (anaSayfaSlaytResmi.isDisplayed()) {
            Assert.assertTrue(true);
            Log4j.info("Ana sayfadaki slayt gösterimi başarılı bir şekilde görüntülendi. Sonuç: Ana sayfa açıldı.");
        } else {
            Assert.fail("Ana sayfa açılmadı veya slayt görüntülenemedi.");
            Log4j.error("Ana sayfa açılmadı veya slayt görüntülenemedi.");
        }
    }
}

File: UIAutomation/Behave/pages/HomePage.py
import random
from selenium.webdriver.common.by import By
from utils.Driver import Driver
from utils.Helper import Helper


class Elements:
    TUM_CEREZLERE_IZIN_VER = (By.XPATH, "//div[@id='cookieseal-banner']//button[text()='Tüm Çerezlere İzin Ver']")
    KATEGORI_MENU_SECIM = (By.XPATH, "//div[@class='mega-menu ']//ancestor::li[@class='menu-header-item  ']")
    KATEGORI_SECIM = (
        By.XPATH, "//li[@class='menu-header-item menu-header-item--active ']//ul[@class='zones-lists-items']//a")
    ANA_SAYFA_LOGO = (By.CSS_SELECTOR, ".main-header-logo")
    ANA_SAYFA_SLAYT_RESMI = (By.CSS_SELECTOR, ".full-width-slider.full-width-slider--horizontal ")
    EL_SEARCH_BUTTON = (By.XPATH, "//input[@value='Search']")


class HomePage:
    def __init__(self):
        self.driver = Driver.get()

    def tum_cerezlere_izin_ver(self):
        Helper.click_element(Elements.TUM_CEREZLERE_IZIN_VER)

    def herhangi_bir_kategori_menu_secim(self):
        random_index = random.randint(0, Helper.getCount(Elements.KATEGORI_MENU_SECIM) - 1)
        print(f"Header bölümündeki menülerden '{Helper.getTexts(Elements.KATEGORI_MENU_SECIM, random_index)}' fareyle üzerine gelindi.")

        Helper.hover_elements(Elements.KATEGORI_MENU_SECIM, random_index)


    def herhangi_bir_kategori_secim(self):
        random_index = random.randint(0, Helper.getCount(Elements.KATEGORI_SECIM) - 1)
        print(f"Header bölümündeki kategorilerden '{Helper.getTexts(Elements.KATEGORI_SECIM, random_index)}' tıklandı.")

        Helper.click_elements(Elements.KATEGORI_SECIM, random_index)


    def ana_sayfaya_git(self):
        Helper.click_element(Elements.ANA_SAYFA_LOGO)

    def ana_sayfa_acildi_mi(self):
        if Helper.is_displayed(Elements.ANA_SAYFA_SLAYT_RESMI):
            print("Ana sayfa açıldı.")
        else:
            raise AssertionError("Ana sayfa açılmadı veya slayt görüntülenemedi.")
